### DISCUSSION TITLE
icc_transform with a CMYK to CMYK device link profile

### DISCUSSION THREAD

**Discussion by jindili** [libvips/libvips#2327] [Q&A]
vips version: vips-8.11.0-Wed Jun  9 18:20:35 UTC 2021
system: Windows 10 64bit

I add some sequence of adjustments layer in Photoshop ( like Curves and Hue/Saturation adjust ) based on a CMYK image ( color profile ps5cmyk.icc)
and export Color Lookup Tables as ICC Profile format, and get two icc profiles:
1_adjust_cmyk.icc
1_adjust_cmykRGB.icc

attached these two icc files in a zip file:
[icc.zip](https://github.com/libvips/libvips/files/6737390/icc.zip)

and run with vips command line：

```
vips icc_transform cmyk.tif cmyk_output.tif 1_adjust_cmyk.icc --input-profile ps5cmyk.icc
```

get error:

```
VipsIcc: ColorSpace mismatch
VipsIcc: Couldn't link the profiles
```

info of the 1_adjust_cmyk.icc:
![image](https://user-images.githubusercontent.com/73845/123888610-76635b00-d986-11eb-9b2e-4b1518552a51.png)

info of the 1_adjust_cmykRGB.icc:
![image](https://user-images.githubusercontent.com/73845/123888646-85e2a400-d986-11eb-86fe-8db272a2b12b.png)

cmyk.tif file used:
[cmyk.zip](https://github.com/libvips/libvips/files/6737425/cmyk.zip)
a simple gradient
![image](https://user-images.githubusercontent.com/73845/123889291-c42c9300-d987-11eb-87f5-bf0dc14febb9.png)

**Comment by jcupitt** [comment]
Hi @jindili,

I get "not found" for your icc.zip, could you upload again?

**Comment by jindili** [comment]
Hi John,
Sorry for that,  I get "not found" too.  Upload it again.

[icc.zip](https://github.com/libvips/libvips/files/6739873/icc.zip)

This link above works fine for me.

**Reply by jcupitt** [reply]
Could you upload `ps5cmyk.icc` as well please?

**Reply by jindili** [reply]
sure, it's here
[ps5cmyk.zip](https://github.com/libvips/libvips/files/6740124/ps5cmyk.zip)

**Comment by jcupitt** [comment]
Ah I see. libvips doesn't support adjustment profiles, it only allows profiles to go between device space and PCS, where PCS is LAB or XYZ.

We'd need to add support for this feature.

**Reply by jindili** [reply]
Thanks John, I'll checkout icc_transform.c and see if it's possible for me to add this feature.
